Hw4Task3/Hw4Task3/Hw4Task3.c

```diff
@@ -0,0 +1,292 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+typedef struct {
+    char name[30];
+    char phone[30];
+} PhoneBookEntry;
+
+// function for reading data from file at the beginning of the program; testOrNot = true => opens test.txt, else opens data.txt
+int readDataFromFile(PhoneBookEntry* notesArray, int currentIndex, bool testOrNot)
+{
+    FILE* notes;
+    if (testOrNot)
+    {
+        notes = fopen("test.txt", "r");
+    }
+    else
+    {
+        notes = fopen("data.txt", "r");
+    }
+    if (notes == NULL)
+    {
+        printf("File not found");
+        return;
+    }
+    char* data[200] = { 0 };
+    int linesRead = 0;
+
+    while (!feof(notes))
+    {
+        char* buffer = malloc(sizeof(char) * 100);
+        const int readBytes = fgets(buffer, 30, notes);
+        if (readBytes < 0)
+        {
+            free(buffer);
+            break;
+        }
+        data[linesRead] = buffer;
+        ++linesRead;
+    }
+    for (int index = 0; index < linesRead - 1; index += 2)
+    {
+        strcpy(notesArray[currentIndex].name, data[index]);
+        strcpy(notesArray[currentIndex].phone, data[index + 1]);
+        ++currentIndex;
+    }
+    fclose(notes);
+    return currentIndex;
+}
+
+// function for adding note (name and phone number) to an array with all notes
+void addNote(PhoneBookEntry* notesArray, char* name, char* phone, int currentIndex)
+{
+    strcpy(notesArray[currentIndex].name, name);
+    strcpy(notesArray[currentIndex].phone, phone);
+    return;
+}
+
+// function for printing all notes in console
+void printAllNotes(PhoneBookEntry* notesArray, int currentIndex)
+{
+    for (int index = 0; index < currentIndex; index++)
+    {
+        printf("Name: %sPhone Number: %s", notesArray[index].name, notesArray[index].phone);
+        printf("-----\n");
+    }
+    return;
+}
+
+// function for searching phone number using name
+int findPhoneNumberUsingName(PhoneBookEntry* notesArray, char* toSearchName, int currentIndex)
+{
+    int indexWithWantedNote = -1;
+    for (int index = 0; index < currentIndex; index++)
+    {
+        if (!strcmp(notesArray[index].name, toSearchName))
```

Comment:
```suggestion
        if (strcmp(notesArray[index].name, toSearchName) == 0)
```
--- всё-таки strcmp возвращает целочисленное значение